src/main/java/com/epam/javacourses2016/task8/SolverTask8.java

```diff
@@ -10,12 +12,59 @@
     /**
      * Проверяет правильность расстановки скобок.
      * Правильная расстановка:
-     *      1) Каждой открывающей скобке соответствует закрывающая того же типа.
-     *      2) Нет пересечения областей, обрамленных скобками.
+     * 1) Каждой открывающей скобке соответствует закрывающая того же типа.
+     * 2) Нет пересечения областей, обрамленных скобками.
+     *
      * @param string Анализируемая строка.
      * @return true - скобки расставлены верно, иначе - false.
      */
     public boolean isNormalBrackets(String string) {
-        return false;
+        Stack<Character> characters = new Stack<>();
+
+        char[] openTypes = new char[]{'(', '[', '{'};
+        char[] closingTypes = new char[]{')', ']', '}'};
```

Comment:
There could be situation like {some text] - and if i ask set .contains() it give me true, but it will be wrong answer. I have to check pairs also {} [] etc, thats why i need to know index. I thought about map, but i feel like it will be slower than array of chars.